src/frontend/src/catalog/system_catalog/pg_catalog/pg_database.rs

```diff
@@ -36,58 +36,32 @@ use crate::catalog::system_catalog::{BuiltinTable, SysCatalogReaderImpl};
 /// ```
 ///
 /// Ref: [`pg_database`](https://www.postgresql.org/docs/current/catalog-pg-database.html)
-pub const PG_DATABASE: BuiltinTable = BuiltinTable {
+pub static PG_DATABASE: LazyLock<BuiltinView> = LazyLock::new(|| BuiltinView {
     name: "pg_database",
     schema: PG_CATALOG_SCHEMA_NAME,
     columns: &[
         (DataType::Int32, "oid"),
         (DataType::Varchar, "datname"),
-        // None
         (DataType::Int32, "datdba"),
-        // 6
         (DataType::Int32, "encoding"),
-        // 'C'
         (DataType::Varchar, "datcollate"),
-        // 'C'
         (DataType::Varchar, "datctype"),
-        // false
         (DataType::Boolean, "datistemplate"),
-        // true
         (DataType::Boolean, "datallowconn"),
-        // -1
         (DataType::Int32, "datconnlimit"),
-        // 1663
         (DataType::Int32, "dattablespace"),
-        // null
         (DataType::Varchar, "datacl"),
     ],
-    pk: &[0],
-};
-
-pub fn new_pg_database_row(id: u32, name: &str) -> OwnedRow {
-    OwnedRow::new(vec![
-        Some(ScalarImpl::Int32(id as i32)),
-        Some(ScalarImpl::Utf8(name.into())),
-        None,
-        Some(ScalarImpl::Int32(6)),
-        Some(ScalarImpl::Utf8("C".into())),
-        Some(ScalarImpl::Utf8("C".into())),
-        Some(ScalarImpl::Bool(false)),
-        Some(ScalarImpl::Bool(true)),
-        Some(ScalarImpl::Int32(-1)),
-        Some(ScalarImpl::Int32(1663)),
-        None,
-    ])
-}
-
-impl SysCatalogReaderImpl {
-    pub fn read_database_info(&self) -> Result<Vec<OwnedRow>> {
-        let reader = self.catalog_reader.read_guard();
-        let databases = reader.get_all_database_names();
-
-        Ok(databases
-            .iter()
-            .map(|db| new_pg_database_row(reader.get_database_by_name(db).unwrap().id(), db))
-            .collect_vec())
-    }
-}
+    sql: "SELECT id AS oid, \
+                 name AS datname, \
+                 owner AS datdba, \
+                 6 AS encoding, \
+                 'C' AS datcollate, \
+                 'C' AS datctype, \
+                 false AS datistemplate, \
+                 true AS datallowconn, \
+                 -1 AS datconnlimit, \
+                 1663 AS dattablespace, \
+                 acl AS datacl FROM rw_catalog.rw_databases"
```

Comment:
Since this PR `datdba` and `datacl` are filled with real values rather than `null`.

What about fixing `datctype` from `C` to `UTF-8` as well?

Reply:
Did you mean fixing `datcollate` an `datctype` from `C` to `en_US.UTF-8` by default? They're determined by the option `--locale` or the server operating system when initialized calling `initdb`. LGTM, I will change it.